src/appleseed/renderer/modeling/material/imaterialfactory.cpp

```diff
@@ -169,4 +169,23 @@ void IMaterialFactory::add_default_tangent_mode_metadata(DictionaryArray& metada
             .insert("default", "uv"));
 }
 
+void IMaterialFactory::add_microfacet_normal_mapping_metadata(foundation::DictionaryArray& metadata)
+{
+    metadata.push_back(
+        Dictionary()
+            .insert("name", "microfacet_normal_mapping")
+            .insert("label", "Microfacet Normal Mapping")
+            .insert("type", "numeric")
+            .insert("min",
+                Dictionary()
+                    .insert("value", "0.0")
+                    .insert("type", "hard"))
+            .insert("max",
+                Dictionary()
+                    .insert("value", "1.0")
+                    .insert("type", "hard"))
+            .insert("use", "optional")
+            .insert("default", "0.0"));
+}
```

Comment:
Adding
```
.insert("visible_if",
                Dictionary()
                    .insert("displacement_method", "normal")));
```
will make this option only be visible if normal mapping is selected.

Reply:
Great hint, thanks!
Now it is done like this.